feature_engine/discretisation/equal_frequency.py

```diff
@@ -81,23 +82,16 @@ def __init__(
         self,
         variables: Union[None, int, str, List[Union[str, int]]] = None,
         q: int = 10,
-        return_object: bool = False,
-        return_boundaries: bool = False,
     ) -> None:
 
         if not isinstance(q, int):
             raise ValueError("q must be an integer")
 
-        if not isinstance(return_object, bool):
-            raise ValueError("return_object must be True or False")
-
-        if not isinstance(return_boundaries, bool):
-            raise ValueError("return_boundaries must be True or False")
-
+        super().__init__(return_object, return_boundaries)
+        
         self.q = q
         self.variables = _check_input_parameter_variables(variables)
-        self.return_object = return_object
-        self.return_boundaries = return_boundaries
+
 
     def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
```

Comment:
if all works as planned, we should be able to delete the transform method from this class, because it would inherit that on the base class.

But be mindful of the dosctrings. In short, we need to copy and paste the docstrings from the transform method in this class to the base class.